### ISSUE TITLE
Form components: conflict with id and name attributes for input

### ISSUE THREAD

**Issue by stefankp** [primer/brand#116]
The current API for `FormControl` relies on the `id` field to assign the necessary `name` attribute for input tags. I would expect to be able to assign a custom `name` attribute to any input, regardless of what `id` was defined for the `FormControl`.

#### This creates an autogenerated `id` and `name`

```JSX
<FormControl>
  <FormControl.Label>Name</FormControl.Label>
  <TextInput />
</FormControl>
```

#### This creates an autogenerated `id` and `name`, ignores the custom `name` set on the input

```JSX
<FormControl>
  <FormControl.Label>Name</FormControl.Label>
  <TextInput name="name" />
</FormControl>
```

#### This sets `id` and `name` to the value `name`

```JSX
<FormControl id="name">
  <FormControl.Label>Name</FormControl.Label>
  <TextInput />
</FormControl>
```

**Comment by rezrah** [comment]
Thanks @stefankp - I've rolled the fix into https://github.com/primer/brand/pull/117

**Comment by rezrah** [comment]
@stefankp - the fix has been merged into `main` but will ship in `0.7.0` release. You can track it here: https://github.com/primer/brand/pull/119
